app/src/pages/clinic.tsx

```diff
@@ -19,9 +20,28 @@ const Clinic: NextPage = () => {
   const handleSearch = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault()
     setSelectedClinic(undefined)
-    const filtered = clinics.filter((clinic) => clinic.zip === search)
 
-    setFilteredClinics(filtered)
+    if (search) {
+      import(
+        `../../public/clinic-output/clinics-zip-code-lookup/${search}.json`
+      )
+        .then(
+          (sortedClinics: { default: { id: number; distance: string }[] }) => {
+            const clinicsWithDetails: typeof clinics = sortedClinics.default
+              .slice(0, 8)
+              .map(
+                (clinic: typeof sortedClinics.default[0]) =>
+                  clinics.find(
+                    (clinicDetails: typeof clinics[0]) =>
+                      clinicDetails.id === clinic.id
+                  ) || clinics[0]
+              )
```

Comment:
```suggestion
                  )
              ).filter(Boolean);
```
...which means we don't need the `||` anymore. I also suggest adding the filter to remove any `undefined` values from the result of the `map` (even though we don't really expect this to happen).

```js
> [undefined, undefined, undefined].filter(Boolean)
[]
```